aten/src/ATen/function_wrapper.py

```diff
@@ -157,14 +166,36 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 #endif
 }
 """)
+
+C10_TENSOR_METHOD_DEFINITION_UNDERSCORE = CodeTemplate("""\
```

Comment:
I'd like to understand if there is a way we can avoid having both `C10_TENSOR_METHOD_DEFINITION` and `C10_TENSOR_METHOD_DEFINITION_UNDERSCORE`, because having it duplicated like this means it's 2x the work if we need to update templates in the future. It looks like the only difference between the templates is `${api_name}` versus `_${api_name}`. Can you just mangle `${api_name}` (or add a `${api_prefix}` variable) and reuse the template in both cases? This applies to the other cases too.